File: madison/frontend/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  category: string;
  price: number;
  isActive: boolean;
  isDeleted: boolean;
  isDangerousDrug: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface UpdateProductDescriptionDto {
  productId: number;
  description: string;
}

export interface ProductListResponse {
  products: Product[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
} 

File: madison/backend/VeterinaryAPI/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace VeterinaryAPI.Models
{
    public class Product
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [StringLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        [StringLength(50)]
        public string Category { get; set; } = string.Empty;
        
        public decimal Price { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public bool IsDeleted { get; set; } = false;
        
        public bool IsDangerousDrug { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? UpdatedAt { get; set; }
    }
} 
